software package search indexer not working after 'rhn-search cleanindex'

In GUI Sofftware -> Package Search are "No packages" for any searches.
"rhn-search cleanindex" dosn't help.
"spacecmd package_search" works fine.

# Additional Information

## Version of Uyuni Server and Proxy (if used)
2020.06-78.2.uyuni

## Details about issue
In GUI Sofftware -> Package Search are "No packages" for any searches.
In /var/log/rhn/search/rhn_search.log are errors:
ERROR - no segments* file found in org.apache.lucene.store.FSDirectory@/var/lib/rhn/search/indexes/package: files:

"rhn-search cleanindex" works without errors and deletes directories under /var/lib/rhn/search/indexes
But only in the directories  "errata", "hwdevice", "server", "serverCustomInfo", "snapshotTag" and "xccdfIdent" are indexes rebuildt. The directory "packages" stays empty.

In /var/log/rhn/search/rhn_search*.log are entires for all indexers except the package-indexer. No errors are logged.

Good advice - I searched in wrong place!  In journal are OutOfMemoryExceptions:

Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: [QuartzScheduler_Worker-8] ERROR org.quartz.core.JobRunShell - Job updateIndex.packages threw an unhandled Exception:
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: java.lang.OutOfMemoryError: Java heap space
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: [QuartzScheduler_Worker-8] ERROR org.quartz.core.ErrorLogger - Job (updateIndex.packages threw an exception.
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: org.quartz.JobExecutionException: com.redhat.satellite.search.index.IndexingException: org.apache.lucene.store.LockObtainFailedException: Lock o>
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at com.redhat.satellite.search.scheduler.tasks.GenericIndexTask.execute(GenericIndexTask.java:94)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: Caused by: com.redhat.satellite.search.index.IndexingException: org.apache.lucene.store.LockObtainFailedException: Lock obtain timed out: Simple>
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at com.redhat.satellite.search.index.IndexManager.createIndex(IndexManager.java:247)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at com.redhat.satellite.search.scheduler.tasks.GenericIndexTask.execute(GenericIndexTask.java:65)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         ... 2 more
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: Caused by: org.apache.lucene.store.LockObtainFailedException: Lock obtain timed out: SimpleFSLock@/var/lib/rhn/search/indexes/hwdevice/write.lock
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.apache.lucene.store.Lock.obtain(Lock.java:85)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.apache.lucene.index.IndexWriter.init(IndexWriter.java:1070)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.apache.lucene.index.IndexWriter.init(IndexWriter.java:1050)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:765)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at com.redhat.satellite.search.index.IndexManager.getIndexWriter(IndexManager.java:391)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at com.redhat.satellite.search.index.IndexManager.createIndex(IndexManager.java:229)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         ... 3 more
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: java.lang.OutOfMemoryError: Java heap space]
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]:         at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Aug 20 14:31:46 rz-mg-lin301 rhn-search[4301]: Caused by: java.lang.OutOfMemoryError: Java heap space

--- a/search-server/spacewalk-search/spacewalk-search.changes
+++ b/search-server/spacewalk-search/spacewalk-search.changes
@@ -1,3 +1,5 @@
+- change default maximum memory to 512 MB, preventing OutOfMemoryError
+
 -------------------------------------------------------------------
 Fri Sep 18 11:43:02 CEST 2020 - jgonzalez@suse.com
 

--- a/search-server/spacewalk-search/src/config/search/rhn_search_daemon.conf
+++ b/search-server/spacewalk-search/src/config/search/rhn_search_daemon.conf
@@ -8,4 +8,4 @@ SEARCH_RHN_CLASSPATH="/usr/share/rhn/classes:/usr/share/nutch-core:/usr/share/nu
 SEARCH_RHN_JARS="/usr/share/rhn/lib/spacewalk-asm.jar:/usr/share/rhn/lib/rhn.jar:/usr/share/rhn/lib/java-branding.jar"
 SEARCH_JARS="/usr/share/rhn/search/lib/*"
 SEARCH_INIT_MEMORY=32
-SEARCH_MAX_MEMORY=96
+SEARCH_MAX_MEMORY=512

